source/text.c

```diff
@@ -413,6 +433,21 @@ void C2D_DrawText(const C2D_Text* text, u32 flags, float x, float y, float z, fl
 					glyphY = y+dispY*cur->lineNo;
 				}
 
+				if (colors != NULL){
+					if(cur->charNo >= colors[lastColorIdx] && cur->charNo < colors[lastColorIdx+2]) {
```

Comment:
Out of bounds access when the last character was in the last colour range, and the current is also in that range.
Needs a check similar to 5 lines below to prevent it.